code/workspaces/web-app/src/reducers/assetRepoReducer.js

```diff
@@ -15,6 +15,18 @@ const initialState = {
   error: null,
 };
 
+export function addNewAssets(oldAssets, newAssets) {
+  if (!oldAssets || oldAssets.length === 0) { return newAssets; }
+  if (!newAssets || newAssets.length === 0) { return oldAssets; }
+
+  const oldAssetIds = oldAssets.map(asset => asset.assetId);
+  const brandNewAssets = newAssets.filter(newAsset => !oldAssetIds.includes(newAsset.assetId));
+  return [
+    ...oldAssets,
+    ...brandNewAssets,
+  ];
+}
```

Comment:
How come this is needed?

Reply:
It's similar to the logic in stacksReducer - for the case where an admin is looking at multiple projects, it combines the available assets in the redux state.

Reply:
That makes sense with the information given in the above comment, thank you for the clarification!